pkg/cli/mirror/publish.go

```diff
@@ -123,19 +123,28 @@ func (o MirrorOptions) Publish(ctx context.Context, cmd *cobra.Command, f kcmdut
 	}
 
 	// Get current metadata info
-	cfg := v1alpha1.StorageConfig{
-		Registry: &v1alpha1.RegistryConfig{
-			ImageURL: metaImage,
-			SkipTLS:  o.DestSkipTLS,
-		},
+	var cfg v1alpha1.StorageConfig
+	if !incomingMeta.SingleUse {
+		cfg = v1alpha1.StorageConfig{
+			Registry: &v1alpha1.RegistryConfig{
+				ImageURL: metaImage,
+				SkipTLS:  o.DestSkipTLS,
+			},
+		}
 	}
-	backend, err := storage.ByConfig(ctx, o.Dir, cfg)
-	if err != nil {
-		return err
+	backend, berr := storage.ByConfig(o.Dir, cfg)
```

Comment:
If a function named `ByConfig` is being called, I would assume that the backend is configured, i.e. `cfg` can be used if valid. Then it would make more sense to handle backend non-configuration in the caller rather than storage, ex.
```go
var backend storage.Backend
if incomingMeta.SingleUse {
	logrus.Warn("metadata has single-use label, using stateless mode")

	tmp, err := os.MkdirTemp("", "")
	if err != nil {
		return err
	}
	cfg := v1alpha1.StorageConfig{
		Local: &v1alpha1.LocalConfig{
			Path: tmp,
		},
	}
	backend, err = storage.ByConfig(o.Dir, cfg)
	if err != nil {
		return err
	}
	defer func() {
		if err := backend.Cleanup(ctx, config.MetadataBasePath); err != nil {
			logrus.Error(err)
		}
	}()
} else {
	cfg := v1alpha1.StorageConfig{
		Registry: &v1alpha1.RegistryConfig{
			ImageURL: metaImage,
			SkipTLS:  o.DestSkipTLS,
		},
	}
	backend, err = storage.ByConfig(o.Dir, cfg)
	if err != nil {
		return err
	}
}
```

Reply:
This solution makes sense to me. My original thought was that I would return a temporary backend if an empty config was supplied, but I see how that can conflict with the idea of `ByConfig` especially as the pool of backend options grows. Reworking this.